tests/cpp/test_matmul_scheduler.cpp

```diff
@@ -3409,4 +3409,74 @@ TEST_F(MatmulSchedulerTest, HSH_NN) {
   EXPECT_FALSE(at::allclose(cg_outputs[0], tref, 1e-5, 1e-5));
 }
 
+// Test a kernel with MmaOp that has inputs like [B, M, K] and [B, K, N]
+// and whose output is [B, M, N, K]
+TEST_F(MatmulSchedulerTest, MmaOpAxisMapping) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  int64_t B = 4, M = 16, N = 32, K = 64;
+  DataType dtype = DataType::Half;
+
+  auto tv0 = makeConcreteTensor({B, M, K}, dtype);
+  auto tv1 = makeConcreteTensor({B, K, N}, dtype);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  // Just doing a gmem->smem copy
+  tv0 = set(tv0);
+  tv0->setMemoryType(MemoryType::Shared);
+  tv1 = set(tv1);
+  tv1->setMemoryType(MemoryType::Shared);
+
+  MmaOp::AxisMapping axis_mapping{
+      /*a_axes=*/{0, 1, -1, 2},
+      /*b_axes=*/{0, -1, 2, 1}};
+
+  auto tv2 =
+      fusedMultiplySum(tv0, tv1, /*axes=*/{-1}, /*init=*/nullptr, axis_mapping);
+
+  fusion.addOutput(tv2);
+
+  std::vector<mma_utils::MatmulPattern> patterns =
+      mma_utils::findMatmulPatterns(&fusion);
+
+  ASSERT_EQ(patterns.size(), 1);
+  const mma_utils::MatmulPattern& pattern = patterns.front();
+
+  IdModel id_model(&fusion);
+  const ValGraph& permissive_graph =
```

Comment:
Let's use broadcast graph instead of permissive graph.

Reply:
I am actually planning to do this, but it means I need to update mma_utils since permissive is still used there, so I will do that in a follow-up.

Reply:
#3397